### PR TITLE
[REF] web: refactor debug menu to use context in env instead of service

### PR DESCRIPTION
**Pull request by sdegueldre**

### CODE REVIEW

File: addons/web/static/src/core/debug/debug_service.js [hunk 1]
```diff
@@ -29,46 +37,70 @@ export const debugService = {
...
+                return () => {
+                    const contexts = activeCategories.get(category) || new Set();
+                    contexts.delete(context);
+                    if (contexts.size === 0) {
+                        activeCategories.delete(category);
+                    }
+                };
+            },
+
+            getItems(env) {
```
Comment (sdegueldre) [inline addons/web/static/src/core/debug/debug_service.js:61]: make async and load rights here

File: addons/web/static/src/core/debug/debug_service.js [hunk 2]
```diff
@@ -1,26 +1,34 @@
...
+        });
     });
-}
+    const [canEditView, canSeeRecordRules, canSeeModelAccess] = await Promise.all(proms);
+    const accessRights = { canEditView, canSeeRecordRules, canSeeModelAccess };
+    return accessRights;
+});
 
-export const debugService = {
+export const debugService = ({ withRoot }) => ({
```
Comment (sdegueldre) [inline addons/web/static/src/core/debug/debug_service.js:28]: use single debugService that creates two instances of managing objects

File: addons/web/static/src/core/debug/debug_service.js [hunk 3]
```diff
@@ -29,46 +37,70 @@ export const debugService = {
...
-                            .then(() => resolve(accessRights))
-                            .catch((error) => {
-                                accessRightsProm = undefined;
-                                reject(error);
-                            });
+            async loadAccessRights() {
+                accessRights = await getAccessRights(orm);
+            },
+
+            activateCategory(category, context) {
```
Comment (sdegueldre) [inline addons/web/static/src/core/debug/debug_service.js:47]: create "default" category

File: addons/web/static/src/core/debug/debug_menu.js [hunk 4]
```diff
@@ -1,92 +1,21 @@
...
-        // @ts-ignore
-        this.beforeOpenDropdown = async () => {
-            if (!this.hasAccessRights) {
-                this.hasAccessRights = true;
-                const accessRights = await this.debugService.getAccessRights();
-                Object.assign(globalContext, { accessRights });
-            }
+        this.debugContext = useCurrentDebugContext();
+        // Needs to be bound to this for use in template
+        this.getElements = async () => {
```
Comment (ged-odoo) [inline addons/web/static/src/core/debug/debug_menu.js:9]: move logic in debugcontext?

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by sdegueldre** [comment]
@robodoo retry

**Pull request closed without merging**